src/Application.vala

```diff
@@ -114,6 +114,17 @@ public class Music.Application : Gtk.Application {
         }
 
         settings.bind ("window-maximized", main_window, "maximized", SettingsBindFlags.SET);
+
+        // Restoring the queue overwrites the last played. So we need to retrieve it before
+        var? uri_last_played = settings.get_string ("uri-last-played");
+
+        // This needs to be done after window is constructed
+        // Else music plays but the queue seems empty
+        PlaybackManager.get_default ().restore_queue ();
+
+        settings.set_string ("uri-last-played", uri_last_played);
+        PlaybackManager.get_default ().restore_last_played ();
```

Comment:
Is there a reason to have this be two separate functions? We always want to restore both the previous queue and the previously played audio right?

Reply:
Oh yeah that makes sense. Thanks for clarifying!